docs/advanced/signal_generators.md

```diff
@@ -0,0 +1,558 @@
+| Sample Rate (S/s) | 0.298k–2.5G | 0.298k–5.0G | "              | "         |
+| Amplitude (V)     | 25m–0.75    | 25m–0.75    | 25m–1.5        | 10m-5.0   |
+| Offset (V)        | -2–2        | -2–2        | -2–2           | -2-2      |
+
+```
+
+#### Sequential, Blocking and Overlapping Commands:
+
+The AWG5200's programming commands are seperated into three seperated categories: Sequential, Blocking, Overlapping.
+The type of command is important to consider as an incorrect order can lead to unintended results.
+
+Sequential commands function as the standard PI commands. They will not start until the previous command has finished. These commands
+tend to be fast and will allow for quick response times even if they are queued in the input buffer.
+
+Blocking commands are very similar to sequential commands. The main difference between these commands is that
+blocking commands often take longer to execute.
+
+```{caution}
+Due to the length of blocking commands, a query may time out when sent if performed immediately after a large series of consecutive blocking commands.
+```
+
+Some commands can perform data analysis on another thread, these are referred to as overlapping commands. They allow any command to be started
+while they are being executed. They cannot begin if the previous command was blocking or sequential, or if the operation complete status register is
+not set.
+
+```{tip}
+Overlapping commands run in parralel with any other command, so placing them first in a sequence is always preferable.
+```
+
+```{tip}
+There are multiple ways of synchronizing overlapping commands. This includes using Operation Complete (OPC) or Wait (WAI) to
+wait for the operation complete to clear in the SESR (Standard Event Status Register). This can also be done using an SRQ (Service Request),
+along with the waiting for trigger bit in the OCR (Operation Condition Register).
+```
+
+```{caution}
+The operation complete register will only wait for the first overlapping command to finish before clearing. This means
+that if multiple overlapping commands are run, then subsequent overlapping commands being finished will be ignored.
+```
+
+```{danger}
+Overlapping commands can cause unintended behavior when performed alongside critical hardware functionality.
+If the AWG5200 is experiencing problems, this may be a cause.
+```
+
+### AWG70KA, AWG70KB
+
+The AWG70K series are signal sources focused on waveform generation which operate on Windows 10.
+They accept communication through USB, LAN and GPIB interfaces.
+
+#### Constraints:
+
+The AWG70K is a special case, where only the direct signal output path is allowed (unless option AC is installed). This means the amplitude is limited,
+and offset is not allowed to be set by default. However, there is a secondary device which allows for DC amplification, the MDC4500-4B. The MDC4500-4B is an amplifier
+which provides an AWG70K the ability to utilize DC offset. It also provides a large range for amplitude.
```

Comment:
```suggestion
which provides an AWG70K with the ability to utilize DC offset. It also provides a large range for amplitude.
```

Reply:
"the ability" is already an adverbial phrases modifying "to utilize" to effect the instrument of the sentence.